### PR TITLE
service_config: update doc for MethodConfig and LoadBalancingConfig

### PR DESCRIPTION
**Pull request by menghanl**
MethodConfig:
 - Allow a default config where service name is not set
 - Clarify validation

LoadBalancingConfig:
 - Clarify validation

### CODE REVIEW

File: grpc/service_config/service_config.proto [hunk 1]
```diff
@@ -36,21 +36,32 @@ option java_outer_classname = "ServiceConfigProto";
 
 // Configuration for a method.
 message MethodConfig {
-  // The names of the methods to which this configuration applies. There must
-  // be at least one name. Each name entry must be unique across the entire
-  // ClientConfig. If the 'method' field is empty, then this MethodConfig
-  // specifies the defaults for all methods for the specified service.
+  // The names of the methods to which this configuration applies.
+  // - MethodConfig without names (empty list) will be skipped.
+  // - Each name entry must be unique across the entire ClientConfig.
```
Comment (markdroth) [inline grpc/service_config/service_config.proto]: s/ClientConfig/ServiceConfig/

(pre-existing problem, but please fix while you're in here)

File: grpc/service_config/service_config.proto [hunk 2]
```diff
@@ -36,21 +36,32 @@ option java_outer_classname = "ServiceConfigProto";
...
+  // - If the 'method' field is empty, this MethodConfig specifies the defaults
+  //   for all methods for the specified service.
+  // - If the 'service' field is empty, the 'method' field must be empty, and
+  //   this MethodConfig specifies the default for all methods (it's the default
+  //   config).
   //
   // For example, let's say that the service config contains the following
```
Comment (markdroth) [inline grpc/service_config/service_config.proto]: Right before this sentence, let's explicitly add the sentence "When determining which MethodConfig to use for a given RPC, the most specific match wins."

File: grpc/service_config/service_config.proto [hunk 3]
```diff
@@ -220,10 +231,15 @@ message XdsConfig {
...
-// In general, all instances of this field should be repeated.
-// Clients will iterate through the list in order and stop at the first
-// policy that they support.  This allows the service config to specify
-// custom policies that may not be known to all clients.
+// In general, all instances of this field should be repeated. Clients will
+// iterate through the list in order and stop at the first policy that they
+// support.  This allows the service config to specify custom policies that may
+// not be known to all clients.
+//
+// - If the config for the first registered policy is invalid, the whole service
```
Comment (markdroth) [inline grpc/service_config/service_config.proto]: We use the term "supported" in the previous paragraph, but we use the term "registered" in these two bullets.  I think either term is fine, but let's be consistent.

File: grpc/service_config/service_config.proto (hunk 1 above)
Comment (menghanl) [reply grpc/service_config/service_config.proto]: Done

File: grpc/service_config/service_config.proto (hunk 2 above)
Comment (menghanl) [reply grpc/service_config/service_config.proto]: Done

File: grpc/service_config/service_config.proto (hunk 3 above)
Comment (menghanl) [reply grpc/service_config/service_config.proto]: Changed to `supported`.

File: grpc/service_config/service_config.proto [hunk 4]
```diff
@@ -36,21 +36,33 @@ option java_outer_classname = "ServiceConfigProto";
...
-  // For a request for MyService/Foo, we will use the second entry, because it
-  // exactly matches the service and method name.
-  // For a request for MyService/Bar, we will use the first entry, because it
-  // provides the default for all methods of MyService.
+  // MyService/Foo will use the third entry, because it exactly matches the
+  // service and method name. MyService/Bar will use the second entry, because
+  // it provides the default for all methods of MyService. AnotherService/Baz
+  // will use the first entry, because it doesn't match the other two.
+  //
+  // In JSON representation, "" the same as null, and not present. The following
```
Comment (ejona86) [inline grpc/service_config/service_config.proto]: s/"" the/"" is the/

Comment (menghanl) [reply grpc/service_config/service_config.proto]: Done.

**Pull request merged**